### DISCUSSION TITLE
Dynamically set validation rules

### DISCUSSION THREAD

**Discussion by testerez** [mantinedev/mantine#3222] [Feature requests]
Hello,
I have an invoice form with `zod` validation:
```ts
  const form = useForm({
    validate: zodResolver(zodSchema),
  })
```
 On this form, the end user can either save as a draft or send the edited invoice.
I'd like to have 2 different zod schemas for each of those actions, so that sending the invoice will required more fields to be filled. 
I was thinking that I could change the validation schema dynamically with something like:
```ts
// I wish I could do this:
form.setValidate(zodResolver(newZodSchema))
```
 But looking at the [source code](https://github.com/mantinedev/mantine/blob/master/src/mantine-form/src/use-form.ts#L46), the `validate` object passed to `useForm` cannot be changed afterward.

Does that sound like a reasonable use case? Or maybe there is another way to achieve this?

**Comment by rtivital** [comment]
Validate function can be dynamic, you can store your resolver in separate state. See example of dynamic function – https://mantine.dev/form/recipes/#form-with-multiple-steps
